### PR TITLE
urequests: Remove commented prints

### PR DESCRIPTION
**Pull request by marctc**

### CODE REVIEW

**Comment by pfalcon** [comment]
Why?

**Comment by marctc** [comment]
- A `print` in a Python library is not a recommended way to printing output.
- Letting dead code/commented code is a bad practice (ex: https://medium.com/@kentcdodds/please-don-t-commit-commented-out-code-53d0b5b26d5f#.4ww1hw66u)
- This piece of code are intented to be deployed in small devices. So every line counts in terms of space and performance.

**Comment by pfalcon** [comment]
> Letting dead code/commented code is a bad practice (ex: https://medium.com/@kentcdodds/please-don-t-commit-commented-out-code-53d0b5b26d5f#.4ww1hw66u)

Sorry, but that's typical commercial-programming b/s. Hire me to maintain micropython-lib, I'll remove all debugging and diagnostic code and will bill you hours to resolve the most trivial of the issues. While I'm doing that in my own free time, I'll spend minutes on that, with all diagnostic at my fingertips. I and anybody else, so please don't obfuscate code for the people of Earth.

> This piece of code are intented to be deployed in small devices. So every line counts in terms of space and performance.

Yes, but not at the expense of the code maintainability.

**Comment by marctc** [comment]
> Sorry, but that's typical commercial-programming b/s
...nuff said. Enjoy your with your print debugging and git ignorance.

Cheers

**Comment by pfalcon** [comment]
git ignorance is when you submit a patch without bothering to look into git log for how a commit message should be formatted, and even ignore the explicit link to "contributing" document shown to everyone creating a pull request.

**Comment by marctc** [comment]
Maybe I'm a git ignorant to, although I read the contributing link (look a the PR title). But I never left dead code on any lib and never answered a contributor in a such disrespectful way that you answered to me.

**Comment by pfalcon** [comment]
I'm sorry again that some people (like me) have a strong opinion on somebody else's opinions which people (like you) like to repeat. Don't take it personally. But uncritically applying "there should be no single line of commented code" sophism and removing important diagnostic code is the same as in construction, putting a known to be soon rusty pipe in the concrete, so a poor customer in few years had to demolish a wall to diagnose and replace it. That's not just "disrespectful", it's a deliberate wrong-doing.

**Comment by marctc** [comment]
@pfalcon I can understand your approach. But if something make a PR with a full of prints or a lot of commented code, will you reject it? Where are the red lines? That's why it's important to follow the "standards".

ITOH, If your lib have the rule to use "commented prints for debugging" its preferably to be properly documented instead having this absurd rant and write comments like "Sorry, but that's typical commercial-programming b/s". I used this lib and I really like it. I only wanted to contribute removing those prints as some developer left it due a mistake. If I knew that commented prints were allowed I won't waste my time making a PR.

**Comment by pfalcon** [comment]
>  But if something make a PR with a full of prints or a lot of commented code, will you reject it? Where are the red lines?

There's no single rule for that, it depends on the context (in micropython-lib's case, on a particular module). As usual, the truth is somewhere in the middle, with a strong bias for "less", that's why your patch removes 2 lines, not "a lot". So, I wouldn't accept a patch with lots of commented code, but with few diagnostic prints - perhaps.

> ITOH, If your lib have the rule to use "commented prints for debugging" its preferably to be properly documented instead having this absurd rant and write comments like "Sorry, but that's typical commercial-programming b/s". I used this lib and I really like it. I only wanted to contribute removing those prints as some developer left it due a mistake. If I knew that commented prints were allowed I won't waste my time making a PR.

As hinted above, there's no single formal rule, and there're myriad of cases arising in the maintenance of heterogeneous codebase, and it's hard to document them all. There's however a "rule of thumb" which is a bit fuzzy and not without exceptions, but applies to any project out there: if it's in the codebase, there's likely a reason for that. The reason may be good or not exactly, but approaching a particular project's case with a simplistic black&white rule like "there shouldn't be commented code" with high probability (IMHO) will lead to not exactly good change (after all, everyone already knows that leaving commented code is "bad", so if they do that, there must be good enough reason for that, right?) 

Summing up: I personally don't think that adding anything to guidelines would help (there're too many cases to describe then), but if you think it would, feel free to open a ticket, suggesting an exact formulation and exact place to put it in https://github.com/micropython/micropython-lib/wiki/ContributorGuidelines
